Add MDM Expansion Docs (#4068)
hapifhir/hapi-fhir: hapi-fhir-docs/src/main/resources/ca/uhn/hapi/fhir/docs/server_jpa_mdm/mdm_expansion.md

This `:mdm` parameter qualifier instructs an interceptor in HAPI fhir to expand the set of resources included in the search by their MDM-matched resources. The two above HTTP requests will return the same result. 

> This behaviour is also supported on the `$everything` operation, via a slightly different mechanism. If you call the operation with `_mdm=true`, then MDM expansion will occur on the resource. For example: 

Reviewer: Do we support _mdm on Group, or all?  What does `/$everything?_mdm=true` mean?

```suggestion
This behaviour is also supported on the `$everything` operation, via a slightly different mechanism. If you call the operation with `_mdm=true`, then MDM expansion will occur on the base Patient instance. For example: 
```

Author: You may be confusing $everything with bulk export. 
$everything is just for patients.

```http request
GET http://example.com:8000/Patient/1/$everything?_mdm=true
```

> The above call will perform an `$everything` operation on Patient/1, and also include all resources referred to by patients linked to Patient/1 via MDM.

Reviewer: ```suggestion
This will first lookup all Patients linked to Patient/1, and then  perform an `$everything` including all resources for these patients.
```
